addons/mail/static/src/core/common/thread_service.js

```diff
@@ -566,7 +566,10 @@ export class ThreadService {
      * @param {import("models").Thread} thread
      * @param {boolean} pushState
      */
-    setDiscussThread(thread, pushState = true) {
+    setDiscussThread(thread, pushState) {
+        if (pushState === undefined) {
+            pushState = thread.localId !== this.store.discuss.thread?.localId;
```

Comment:
Nitpicking: `thread.notEq(this.store.discuss.thread)` is better than comparison on `localId`.